### ISSUE TITLE
cb.lambda for makeMBOInfillCriterionCB should be (also) function

### ISSUE THREAD

**Issue by jakob-r** [mlr-org/mlrMBO#344]
Would make many things really nice. Then I could define
```r
crit.cb.exp5 = makeMBOInfillCriterionCB(cb.lambda = function(n)  rexp(n, 1/5))
```

**Comment by ja-thomas** [comment]
Maybe I am missing something, but what's the use case for that?

automatically set parameters problem dependent?

**Comment by berndbischl** [comment]
the semantic seem unlcear to me. a function of "what"? what is n here?

**Comment by jakob-r** [comment]
A function that generates one or n lambda values. This way we could get rid of a lot of some code that sets the lambda value to some random number and at the same time we would give the user full control of the randomness without much code needed.

Side note: You could argue that I can just write my own infill function but at the moment it is complicated to pass your own Infill function to a multipoint propsal.

**Comment by berndbischl** [comment]
Yes but again: please "document" what would happen. Just saying n lambda values are spawned  is not enough. What are they used for?

**Comment by berndbischl** [comment]
I think the main question is for what mbo variants would that be good. 
Parallel lcb obviously. Anything else?

And the function is basically an rng for lambda. Instead of a constant

**Comment by jakob-r** [comment]
Could also be good for sequential MBO, since _maybe_ a random lambda can save us from getting stuck sometimes. Of course you could also directly add other arguments next to the _n_. Maybe a _p_ for the progress and it would be a value between 0 and 1... just fantasizing...

**Comment by berndbischl** [comment]
if that helps to improve the code base, do a PR
but lets merge this before #347

**Comment by jakobbossek** [comment]
Agree with @jakob-r. This seems to be a nice modification 👍
